README.md

```diff
@@ -1,17 +1,71 @@
+# GA4GH service-info specification v1.0.0 [![](https://travis-ci.org/ga4gh-discovery/service-info.svg?branch=develop)](https://travis-ci.org/ga4gh-discovery/service-info) [![](https://img.shields.io/badge/license-Apache%202-blue.svg)](https://raw.githubusercontent.com/ga4gh-discovery/service-info/develop/LICENSE)
 
-## How to view
+Service discovery is at the root of any computational workflow using web-based APIs. Traditionally, this is hard-coded into workflows, and discovery is a manual process. Service-info provides a way for an API to expose a set of metadata to help discovery and aggregation of services via computational methods. It also allows a server/implementation to describe its capabilities and limitations.
 
-Use [Swagger Editor](https://editor.swagger.io/) to [view the specification](https://editor.swagger.io/?url=https://raw.githubusercontent.com/ga4gh-discovery/service-info/develop/service-info.yaml).
+This document is intended to be used by service-info implementors and consumers. 
 
-## How to test
+# Specification
 
-Use [Swagger Validator Badge](https://github.com/swagger-api/validator-badge) to validate the specification.
+Service-info is described in our [OpenAPI specification](./service-info.yaml), which can be visualised using [Swagger Editor](https://editor.swagger.io/?url=https://raw.githubusercontent.com/ga4gh-discovery/service-info/develop/service-info.yaml).
+
+## Design Principles
+
+Service-info is intended to be a lightweight single payload response service to deliver essential metadata to enable service consumers. It should not be computationally expensive to generate and serve this endpoint. As such, the only response is a single successful payload. All other error responses should be handled as the service implementor sees fit. Clients should only process requests, which return with a successful HTTP response code.
+
+## Essentials
+
+All API invocations are made to a configurable HTTP(S) endpoint, receive HTTP headers, and return text or other allowed formatting as requested by the client. Successful requests result with HTTP status code 200 and have the appropriate text encoding in the response body. The client and server may mutually negotiate HTTP/2 upgrade using the standard mechanism.
+
+## Internet Media Types Handling
+
+When responding to a request a server MUST use the fully specified media type for that endpoint. When determining if a request is well-formed, a server MUST allow a internet type to degrade like so:
+
+- application/json; charset=utf-8
+  - application/json
+
+No vendor specific description has been given here as service-info intends should be incorperated into them.
+
+### HTTP Response Codes
```

Comment:
Is this useful? This is just a subset of general HTTP stuff so I wonder if we should either remove it (if not useful), or put it in the YAML file and prescribe the error object (if useful). WDYT?

Reply:
It's such a simple endpoint but in the past I have queried specifications that didn't firm up the error responses. So either we should be really specific in the spec (and so remove this) or leave this in place

Reply:
OK, let's leave this in place for now, and make a decision later. I filed #25 for this.